core/capabilities/webapi/target/target_test.go

```diff
@@ -0,0 +1,242 @@
+	connector        *gcmocks.GatewayConnector
+	lggr             logger.Logger
+	config           Config
+	connectorHandler *ConnectorHandler
+	capability       *Capability
+}
+
+func setup(t *testing.T) testHarness {
+	registry := registrymock.NewCapabilitiesRegistry(t)
+	connector := gcmocks.NewGatewayConnector(t)
+	lggr := logger.Test(t)
+	config := Config{
+		RateLimiter: common.RateLimiterConfig{
+			GlobalRPS:      100.0,
+			GlobalBurst:    100,
+			PerSenderRPS:   100.0,
+			PerSenderBurst: 100,
+		},
+	}
+	connectorHandler, err := NewConnectorHandler(connector, config, lggr)
+	require.NoError(t, err)
+
+	capability, err := NewCapability(config, registry, connectorHandler, lggr)
+	require.NoError(t, err)
+
+	return testHarness{
+		registry:         registry,
+		connector:        connector,
+		lggr:             lggr,
+		config:           config,
+		connectorHandler: connectorHandler,
+		capability:       capability,
+	}
+}
+
+func capabilityRequest(t *testing.T) capabilities.CapabilityRequest {
+	data := map[string]string{
+		"key": "value",
+	}
+	jsonData, err := json.Marshal(data)
+	require.NoError(t, err)
+	encoded := base64.StdEncoding.EncodeToString(jsonData)
+	targetInput := map[string]any{
+		"url":     "http://example.com",
+		"method":  "POST",
+		"headers": map[string]string{"Content-Type": "application/json"},
+		"body":    encoded,
+	}
+	inputs, err := values.NewMap(targetInput)
+	require.NoError(t, err)
+	wfConfig, err := values.NewMap(map[string]interface{}{
+		"timeoutMs": 1000,
+		"schedule":  SingleNode,
+	})
+	require.NoError(t, err)
+
+	return capabilities.CapabilityRequest{
+		Metadata: capabilities.RequestMetadata{
+			WorkflowID:          workflowID1,
+			WorkflowExecutionID: workflowExecutionID1,
+		},
+		Inputs: inputs,
+		Config: wfConfig,
+	}
+}
+
+func gatewayResponse(t *testing.T, msgID string) *api.Message {
+	headers := map[string]string{"Content-Type": "application/json"}
+	body := []byte("response body")
+	responsePayload, err := json.Marshal(webcapabilities.TargetResponsePayload{
+		StatusCode: 200,
+		Headers:    headers,
+		Body:       body,
+		Success:    true,
+	})
+	require.NoError(t, err)
+	return &api.Message{
+		Body: api.MessageBody{
+			MessageId: msgID,
+			Method:    webcapabilities.MethodWebAPITarget,
+			Payload:   responsePayload,
+		},
+	}
+}
+
+func TestRegisterUnregister(t *testing.T) {
+	th := setup(t)
+	ctx := testutils.Context(t)
+
+	regReq := capabilities.RegisterToWorkflowRequest{
+		Metadata: capabilities.RegistrationMetadata{
+			WorkflowID:    workflowID1,
+			WorkflowOwner: owner1,
+		},
+	}
+	err := th.capability.RegisterToWorkflow(ctx, regReq)
+	require.NoError(t, err)
+
+	err = th.capability.UnregisterFromWorkflow(ctx, capabilities.UnregisterFromWorkflowRequest{
+		Metadata: capabilities.RegistrationMetadata{
+			WorkflowID:    workflowID1,
+			WorkflowOwner: owner1,
+		},
+	})
+	require.NoError(t, err)
+}
+
+func TestCapability_Execute(t *testing.T) {
```

Comment:
what about invalid workflow id, workflow execution id, and other error scenarios?

Reply:
Will add those scenarios